vignettes/basic-principles.Rmd

```diff
@@ -27,133 +42,93 @@ In practice, this is done by labeling each record in the data with three additio
-                  timestamp = as.POSIXct("2020-01-01 11:00:00"))
-}))
-```
 
+We can then access out data using the `get_table` function, and include information on data validity period using `include_slice_info = TRUE`:
 ```{r example_1_results}
-# We can access our data using the `get_table` function
-print(get_table(conn, "mtcars"))
+get_table(conn, "mtcars")
 
-# And we can see the time-keeping if we set `include_slice_info = TRUE`
-print(get_table(conn, "mtcars", include_slice_info = TRUE))
+get_table(conn, "mtcars", include_slice_info = TRUE)
 ```
+Note that where e.g. `dplyr::tbl` requires a more exact specification of the table identity (`tbl(conn, DBI::Id(table = "mtcars"))`), `get_table` will parse any character to a `DBI::Id` object input using `SCDB::id`.
+
+The following day, the current data is now the first five rows of our example data.
+We then store this data in the database using update_snapshot:
 
 ```{r example_2}
 # Let's say that the next day, our data set is now the first 5 of our example data
 data <- head(example_data, 5)
-```
 
-```{r update_snapshot_2, eval = FALSE}
-# We then store these data in the database using update_snapshot
 update_snapshot(.data = data,
                 conn = conn,
                 db_table = "mtcars", # the name of the DB table to store the data in
                 timestamp = as.POSIXct("2020-01-02 12:00:00"))
 ```
-```{r update_snapshot_2_hidden, results = 'hide', echo = FALSE}
-# We then store these data in the database using update_snapshot
-invisible(capture.output({
-  update_snapshot(.data = data,
-                  conn = conn,
-                  db_table = "mtcars", # the name of the DB table to store the data in
-                  timestamp = as.POSIXct("2020-01-02 12:00:00"))
-}))
-```
-
-```{r example_2_results}
+We can again use the `get_table` function to see the latest available data, including time-keeping with `include_slice_info = TRUE`:
+```{r example_2_results_a}
+get_table(conn, "mtcars")
 
-# We again use the `get_table` function and see the latest available data
-print(get_table(conn, "mtcars"))
-
-# And we can see the time-keeping if we set `include_slice_info = TRUE`
-print(get_table(conn, "mtcars", include_slice_info = TRUE))
+get_table(conn, "mtcars", include_slice_info = TRUE)
+```
 
-# Since our data is time-versioned, we can recover the data from the day before
-print(get_table(conn, "mtcars", slice_ts = "2020-01-01 11:00:00"))
+Since our data is time-versioned, we can recover the data from the day before
+```{r example_2_results_b}
+get_table(conn, "mtcars", slice_ts = "2020-01-01 11:00:00")
 ```
 
+On day 3, we imagine that we have the same 5 records, but one of them is altered
+
 ```{r example_3}
-# On day 3, we imagine that we have the same 5 records, but one of them is altered
 data <- head(example_data, 5) |>
   dplyr::mutate(hp = ifelse(dplyr::row_number() == 1, hp / 2, hp))
```

Comment:
Not part of the changes I know, but this could be made more readable if you write:
```suggestion
  dplyr::mutate(hp = ifelse(car == "Mazda RX4", hp / 2, hp))
```